src/pad/src/ICeWall.cpp

```diff
@@ -589,6 +589,320 @@ void ICeWall::placePad(odb::dbMaster* master,
   placeInstance(row, snapToRowSite(row, location), inst, orient.getOrient());
 }
 
+void ICeWall::placePads(const std::vector<odb::dbInst*>& insts, odb::dbRow* row)
```

Comment:
This is overly long.  I think the lambdas here don't improve the readability and should be broken out.

Reply:
removed the lambdas and split the code into 2 functions, `placePadsUniform` (very short, but to be consistent) and `placePadsBumpAligned`